### PR TITLE
feat(topology): export metrics describing topology and changes

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This exports the current topology as metrics. I've not added a dashboard yet and we might need additional metrics but this should already help in debugging.

I've also included a somewhat unrelated change to ensure that we log failures.

Relates to #15172

### CODE REVIEW

File: topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java [hunk 1]
```diff
@@ -213,11 +213,11 @@ private void simulateTopologyChange(
   }
 
   private void failFuture(final ActorFuture<?> future, final Throwable error) {
+    LOG.error("Failed to handle topology request", error);
```
Comment (deepthidevaki) [inline topology/src/main/java/io/camunda/zeebe/topology/changes/TopologyChangeCoordinatorImpl.java:216]: :wrench: A warning might be enough. Sometimes the errors are due to user error, like invalid request or new request when there is an ongoing change.

File: topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          .name(NAMESPACE)
+          .name("cluster_changes_started_at")
+          .help("Time at which the current topology change was started")
+          .register();
+  private static final Gauge COMPLETED_AT =
+      Gauge.build()
+          .namespace(NAMESPACE)
+          .name("cluster_changes_completed_at")
+          .help("Time at which the current topology change was completed")
+          .register();
```
Comment (deepthidevaki) [inline topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java:70]: :thought_balloon: Do we really need these ( started_at and completed_at) in the metrics? Or just a metrics for duration of how long a topology change lasted? We can always query the actuator to find the actual timestamps.

**Review by deepthidevaki — APPROVED** [review]
:rocket: 

:question: I remember some time ago @npepinpe  introduced micrometer as a facade for the metrics. Should we start using micrometer already instead of prometheus directly?

File: topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java (hunk 2 above)
Comment (oleschoenburg) [reply topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java:70]: I think I'll remove those for now :+1:

**Comment by oleschoenburg** [comment]
bors merge

**Comment by npepinpe** [comment]
Would be cool, but the first step is injecting it. I'm not sure where the cluster module is built - if it's directly from dist, then easy peasy, shouldn't be an issue. If it's from the broker, then first we need to get the `MeterRegistry` or `ObservationRegistry` in first before we can do anything, and then pass it along. I understand it's not something most want to just tack on to their PR :sweat_smile:

**Comment by oleschoenburg** [comment]
bors retry

**Pull request merged**